### PR TITLE
remove references to "Silex"

### LINKED ISSUE #40681: Maybe it's time to remove references to "Silex" ?
**Symfony version(s) affected**: 4.4

**Description**
I think we needlessly mention 'Silex' because it is not supported, or better replace to "Silex" to "micro-framework"

"Silex" mentioned:
- src/Symfony/Component/HttpKernel/README.md
- src/Symfony/Component/HttpFoundation/Session/Storage/MockFileSessionStorage.php
 

**How to reproduce**

**Possible Solution**

**Additional context**

### PR DESCRIPTION
**Pull request by a1812**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #40681
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/Session/Storage/MockFileSessionStorage.php [hunk 1]
```diff
@@ -114,7 +114,7 @@ public function save()
             $this->data = $data;
         }
 
-        // this is needed for Silex, where the session object is re-used across requests
+        // this is needed for micro-framework, where the session object is re-used across requests
         // in functional tests. In Symfony, the container is rebooted, so we don't have
         // this issue
```
Comment (Nyholm) [inline src/Symfony/Component/HttpFoundation/Session/Storage/MockFileSessionStorage.php]:
```suggestion
        // this is needed when the session object is re-used across multiple requests
        // in functional tests. 
```

File: src/Symfony/Component/HttpKernel/README.md [hunk 2]
```diff
@@ -3,7 +3,7 @@ HttpKernel Component
 
 The HttpKernel component provides a structured process for converting a Request
 into a Response by making use of the EventDispatcher component. It's flexible
-enough to create a full-stack framework (Symfony), a micro-framework (Silex) or
+enough to create a full-stack framework (Symfony), a micro-framework (Symfony) or
 an advanced CMS system (Drupal).
```
Comment (Nyholm) [inline src/Symfony/Component/HttpKernel/README.md]:
```suggestion
enough to create full-stack frameworks, micro-frameworks or advanced CMS system like Drupal.
```

**Review by Nyholm — COMMENTED** [review]
Thank you. 

I agree that Silex references should not be around. Could we reword the comments a bit? I did a suggestion but Im not sure that is the best solution.

File: src/Symfony/Component/HttpKernel/README.md [hunk 3]
```diff
@@ -3,8 +3,7 @@ HttpKernel Component
 
 The HttpKernel component provides a structured process for converting a Request
 into a Response by making use of the EventDispatcher component. It's flexible
-enough to create a full-stack framework (Symfony), a micro-framework (Silex) or
-an advanced CMS system (Drupal).
+enough to create full-stack frameworks, micro-frameworks or advanced CMS system like Drupal.
```
Comment (stof) [inline src/Symfony/Component/HttpKernel/README.md]: `systems` should also be plural, like previous items in the list.

**Review by stof — APPROVED** [review]

**Review by Nyholm — APPROVED** [review]

**Comment by Nyholm** [comment]
Thank you

**Pull request merged**